netlogo-gui/src/main/window/CompilerManager.scala

```diff
@@ -171,12 +171,12 @@ class CompilerManager(val workspace: AbstractWorkspace,
     // this method again. - ST 7/7/06
     if (!isLoading) {
       val proceed = compileProcedures()
-      if (proceed) {
-        world.realloc()
-        world.rememberOldProgram()
-        setGlobalVariables() // also updates constraints
-        compileWidgets()
-      } else {
+      world.realloc()            // reallocates widgets
+      world.rememberOldProgram() // updates world dialect, which is
+                                 // important for 3D to 2D and vice versa
+      setGlobalVariables() // updates constraints
+      compileWidgets()     // update the widget once the compile is done
```

Comment:
If these comments are helpful for the people reading the code, maybe pull them out into a separate chunk explaining things more generally, then sign and date them.